readinglist/utils.py

```diff
@@ -24,3 +31,39 @@ def native_value(value):
 
 def Enum(**enums):
     return type('Enum', (), enums)
+
+
+class TimeStamper(object):
+    """A helper to generate accurate timestamps.
+
+    It increments a fraction in case prevent two simultaneous calls to return
+    the same timestamp.
+
+    This is mainly used for in-memory backend. Ideally, this task should rely
+    on storage backend transaction capabilities to guarantee timestamp unicity
+    in a multiprocess environment.
+
+    .. note:
+
+        Original comes from http://stackoverflow.com/a/157711
+    """
+    fraction = decimal.Decimal('0.001')
+
+    def __init__(self):
+        self.lock = threading.Lock()
```

Comment:
this lock can't work on a server app because we might deploy several processes behind nginx.